src/dropdown/dropdown.ts

```diff
@@ -82,28 +84,47 @@ export class NgbDropdown {
   /**
    * @internal
    */
-  closeFromOutside() {
+  closeFromOutsideClick($event) {
+    if (this.autoClose && !this._isEventFromToggle($event)) {
+      this.close();
+    }
+  }
+
+  /**
+   * @internal
+   */
+  closeFromOutsideEsc() {
     if (this.autoClose) {
       this.close();
     }
   }
+
+  /**
+   * @internal
+   */
+  set toggleElement(toggleElement: any) { this._toggleElement = toggleElement; }
+
+  private _isEventFromToggle($event) { return $event.target === this._toggleElement; }
```

Comment:
Nope. I want to precisely capture clicks on a toggle button. Changing this to `this._toggleElement.contains($event.target)` would also capture menu items clicks which is not what we want.